packages/odyssey-storybook/test/testcafe/framework/pageobjects/buttonPageObject.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {baseUrl} from "../mocks/baseUrl";
+import XPathSelector from "../xpath-selector";
+
+const id = 'components-button'
+const viewMode = 'story'
+const buttonXPathSelector = '//div[@id=\'root\']//Button';
+
+class Button {
+  private readonly type: string;
+  private readonly children: string;
+  private readonly isDisabled: boolean;
+  private readonly size: string;
+  private readonly isWide: boolean;
+  private readonly variant: string;
+
+
+  constructor(
+    type= 'primary',
+    children = 'Button label',
+    isDisabled= false,
+    size= 's',
+    isWide = false,
+    variant = 'primary') {
+    this.type = type;
+    this.children = children;
+    this.isDisabled = isDisabled;
+    this.size = size;
+    this.isWide = isWide;
+    this.variant = variant;
+  }
+
+  iFrameUrl() {
+    return (
+      () => {
+        const baseURL = new URL(baseUrl.concat('/iframe.html'));
+
+        baseURL.searchParams.set('id', id + '--' + this.type)
+        baseURL.searchParams.set('viewMode', viewMode)
+        baseURL.searchParams.set('args',
+          'children:'.concat(this.children) + ';' +
+          'size:'.concat(this.size) + ';' +
+          'disabled:'.concat(String(this.isDisabled)) + ';' +
+          'wide:'.concat(String(this.isWide)) + ';' +
+          'variant:'.concat(this.variant)
+        )
+        return baseURL.toString();
+      }
+    ) ()
+  }
+}
+
+export function buttonBuilder() {
```

Comment:
I think using the builder pattern will make it easier to construct the object (in this case the button) in the tests, given that the objects can have many properties. 

Question: ts is giving me errors without a lot of description for the current way of defining the builder. Any suggestions on how I can resolve it without having to ignore it? 

![image](https://user-images.githubusercontent.com/73128385/167942465-0aec16a5-0465-46c4-81f2-1e91fb927812.png)

Reply:
I'm honestly unsure why we have this child at all. Are we setting props on a React button? That doesn't make sense in the context of TestCafe. I'd leave any tight-coupling with React outta here. Most likely, Odyssey team has Jest for component testing already.

Reply:
@aliparsa-okta I believe the problem here is that TS has no understanding of the type for `this` here. We could perhaps do something open ended like this to say the object we're mutating via `this.type = '...'` or `this.isDisabled = '...'` expected any string property:

```
const schema: Record<string, string> = {}
return Object.assign(schema, {
  setType (type: string) {
    this.type = type
    return this
  }
    // ... etc
})
```

Reply:
Awesome! I'll use that.